### DISCUSSION TITLE
Property not known: "directUrl".

### DISCUSSION THREAD

**Discussion by Schmell** [prisma/prisma#20046] [Q&A] [answered]
This one threw me for a loop as it was working. 
This property is included in the Vercel connection  and again it was working until I updated to 4.16.1.
when i would generate in 4.16.1 it would throw an error, but it was very vague
`Provided object is not a type literal: runtime.Types.DefaultSelection<EventPayload>`
after a lot of searching and head scratching i rolled prisma back to 4.7.0 (what I updated from) and instead of that vague error I got.
`error: Property not known: "directUrl". -->  ........`
so of course getting rid of this property makes the generate work, but this `directUrl` property is still in the Prisma docs so I don't understand why its failing.
I am also unsure how critical this property is to using Prisma with Vercel although everything seems to be working currently

**Comment by janpio** [comment, marked as answer]
`directUrl` was added after 4.7.0, in 4.10.0 to be exact. Downgrading to that version will cause that it is not available anymore.

You should always use the latest Prisma version, so better open a new discussion about your original problem and let us help you figure out what was causing that error.

**Reply by nurul3101** [reply]
Hey @Schmell 👋 

Did you have a chance to look at Jan's response?

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Schmell** [reply]
So if that's the case. then the first error message when using 4.16.2 is still a valid question.
`Provided object is not a type literal: runtime.Types.DefaultSelection<EventPayload>`
What does this mean? 
I still can't seem to get 4.16.1 or 4.16.2 to work. I still get that error and therefore cannot generate and therefore cannot host because vercel needs to run generate

**Reply by Schmell** [reply]
so it does generate on 4.10.0

**Reply by Schmell** [reply]
I guess i will make a new discussion
